src/main/java/Program.java

```diff
@@ -0,0 +1,162 @@
+        if (!eventTask.contains("/at")) {
+            System.out.println(PRINT_ERROR_MESSAGE);
+            System.out.println(LINE_BREAK_SINGLE);
+            return;
+        }
+
+        String description = eventTask.substring(EVENT_INDEX, eventTask.indexOf('/')).trim();
+        String at = eventTask.substring(eventTask.indexOf("/at") + TASK_STRING_OFFSET).trim();
+        Event newEventTask = new Event(description, at);
+
+        listTasks[numItems] = newEventTask;
+        numItems++;
+
+        System.out.println(TASK_ADDED_DONE_TEXT);
+        System.out.println(newEventTask.toString());
+        printTotalTasks();
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.print(ENTER_COMMAND_TEXT);
+    }
+
+    //function to add a new todo task
+    public static void addToDoTask(String toDoTask) {
+        System.out.println(LINE_BREAK_SINGLE);
+        String description;
+
+        if (toDoTask.contains("todo")) {
+            description = toDoTask.substring(TODO_INDEX).trim();
+        } else {
+            description = toDoTask.trim();
+        }
+
+        ToDo newTask = new ToDo(description);
+        listTasks[numItems] = newTask;
+        numItems++;
+
+        System.out.println(TASK_ADDED_DONE_TEXT);
+        System.out.println(newTask.toString());
+        printTotalTasks();
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.print(ENTER_COMMAND_TEXT);
+    }
+
+    //function to list all tasks currently saved by the user
+    public static void listAllTasks() {
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.println("Displaying all items saved:");
+        if (numItems == 0) {
+            System.out.println("No items found...Add some items now!");
+        }
+        for (int i = 0; i < numItems; i++) {
+            System.out.println(i + 1 + ": " + listTasks[i].toString());
+        }
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.print(ENTER_COMMAND_TEXT);
+    }
+
+    //function to mark individual tasks as done
+    public void executeDoneTask(String task) {
+        int taskNum = Integer.parseInt(task.substring(task.indexOf(' ') + 1));
+        if (taskNum > this.getNumItems() || taskNum <= 0) {
+            System.out.println(DONE_TASK_ERROR_MESSAGE);
+        } else {
+            listTasks[taskNum - 1].markAsDone();
+            System.out.println(DONE_TASK_SUCCESS_MESSAGE);
+            System.out.println(listTasks[taskNum-1].toString());
+        }
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.print(ENTER_COMMAND_TEXT);
+    }
+
+    //function to exit program
+    public void executeBye() {
+        this.setCanTerminateHal(true);
+    }
+
+    public static void printTotalTasks() {
+        System.out.println("You now have " + numItems + " task(s) in your list!");
+    }
+
+    public Boolean getCanTerminateHal() {
+        return canTerminateHal;
+    }
+
+    public void setCanTerminateHal(boolean canTerminateHal) {
```

Comment:
I like how you used a function to check if you can terminate the program. Makes the code more readable